tests/unit/modules/test_roles.py

```diff
@@ -0,0 +1,304 @@
+        privilege_group_descriptions
+    )
+    fake_get_service_instance.return_value.RetrieveContent.return_value.authorizationManager.privilegeList = (
+        privileges_list
+    )
+    fake_get_service_instance.return_value.RetrieveContent.return_value.authorizationManager.roleList = [
+        vmomi_old
+    ]
+
+    def _add_mock(name, privIds):
+        assert name == vmomi_add.name
+        privileges = list(vmomi_add.privilege)
+        privileges.sort()
+        privIds.sort()
+        assert privileges == privIds
+
+    fake_get_service_instance.return_value.RetrieveContent.return_value.authorizationManager.AddAuthorizationRole = (
+        _add_mock
+    )
+
+    def _update_mock(roleId, newName, privIds):
+        assert roleId == 1101
+        assert newName == vmomi_update.name
+        privileges = list(vmomi_update.privilege)
+        privileges.sort()
+        privIds.sort()
+        assert privileges == privIds
+
+    fake_get_service_instance.return_value.RetrieveContent.return_value.authorizationManager.UpdateAuthorizationRole = (
+        _update_mock
+    )
+
+    with patch("pyVmomi.vmodl.query.PropertyCollector.ObjectSpec", autospec=True) as fake_obj_spec:
+        yield request.param["old"], request.param["update"], request.param["add"]
+
+
+def test_find_roles(mocked_roles_data, fake_service_instance):
+    _, service_instance = fake_service_instance
+    current_data, _, _ = mocked_roles_data
+    ret = security_roles.find(
+        role_name=current_data[0]["role"],
+        service_instance=service_instance,
+        profile="vcenter",
+    )
+    # comparing 2 dict with '==' fails, because of inner lists with different orders, use drift.drift_report
+    assert drift.drift_report(ret[0], current_data[0]) == {}
+
+
+def test_update_role(mocked_roles_data, fake_service_instance):
+    _, service_instance = fake_service_instance
+    old_role, update_role, _ = mocked_roles_data
+
+    ret = security_roles.find(
+        role_name=old_role[0]["role"],
+        service_instance=service_instance,
+        profile="vcenter",
+    )
+    # comparing 2 dict with '==' fails, because of inner lists with different orders, use drift.drift_report
+    assert drift.drift_report(ret[0], old_role[0]) == {}
+
+    # update existing policy
+    ret = security_roles.save(
+        role_config=update_role[0],
+        service_instance=service_instance,
+        profile="vcenter",
+    )
+
+    assert ret["status"] == "updated"
+
+
+def test_add_role(mocked_roles_data, fake_service_instance):
+    _, service_instance = fake_service_instance
+    old_role, _, add_role = mocked_roles_data
+
+    ret = security_roles.find(
+        role_name=old_role[0]["role"],
+        service_instance=service_instance,
+        profile="vcenter",
+    )
+    # comparing 2 dict with '==' fails, because of inner lists with different orders, use drift.drift_report
+    assert drift.drift_report(ret[0], old_role[0]) == {}
```

Comment:
**suggestion** I'd like to see a test in here that just makes sure there is stuff in this drift report. It's entirely possible for an erroneous mock to make this pass all the time.

Maybe below before line 304, assert `drift.driftt_report(ret[0], old_role[0]) == expected_drift`, something like that.